kyuubi-server/src/main/scala/org/apache/kyuubi/operation/KyuubiOperation.scala

```diff
@@ -156,8 +157,11 @@ abstract class KyuubiOperation(session: Session) extends AbstractOperation(sessi
   override def shouldRunAsync: Boolean = false
 
   override def setState(newState: OperationState): Unit = {
+    MetricsSystem.tracing { ms =>
+      ms.decCount(MetricRegistry.name(OPERATION_STATE, opType, state.toString.toLowerCase))
+      ms.incCount(MetricRegistry.name(OPERATION_STATE, opType, newState.toString.toLowerCase))
```

Comment:
so the terminal states are cumulative, but the intermediate ones are not

Reply:
yes

Reply:
if so, you need to document it clear